modules/dnn/src/layers/einsum_layer.cpp

```diff
@@ -541,7 +541,7 @@ class LayerEinsumImpl CV_FINAL : public EinsumLayer
                 // Use either the preprocessed inputs (if it is available) or the corresponding raw inputs
                 result = pairwiseOperandProcess(!result.empty() ? result : rawInputs[0],
                                                 !result.empty() ? tmpResult : homogenizedInputDims[0],
-                                                !preProcessedInputs[input].empty() ? preProcessedInputs[input] : rawInputs[input],
+                                                (!preProcessedInputs.empty() && !preProcessedInputs[input].empty()) ? preProcessedInputs[input] : rawInputs[input],
```

Comment:
The changes are not in OpenVINO part. How problem is reproduced?

Reply:
Before these changes, 0D test case would fail on OpenCV build with the openvino 2024 backend. @asmorkalov noticed it on his testing units.

Reply:
I think it fails not only on OpenVINO backend because of access to 0th element from empty vector. Just reproduced on OV backend. So I suggest rename a PR, because the change is generic.